packages/components/pie-modal/src/index.ts

```diff
@@ -44,7 +49,20 @@ export class PieModal extends RtlMixin(LitElement) {
         this.addEventListener('click', (event) => this._handleDialogLightDismiss(event));
     }
 
+    connectedCallback () : void {
+        super.connectedCallback();
+        document.addEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.addEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+    }
+
+    disconnectedCallback () : void {
+        document.removeEventListener(ON_MODAL_OPEN_EVENT, this._handleModalOpened.bind(this));
+        document.removeEventListener(ON_MODAL_CLOSE_EVENT, this._handleModalClosed.bind(this));
+        super.disconnectedCallback();
+    }
+
     firstUpdated (changedProperties: DependentMap<ModalProps>) : void {
+        this._dialog?.addEventListener('cancel', (event) => this._handleDialogCancelEvent(event));
```

Comment:
Would the dialog event bubble up to `this`? If that's the case you could potentially move this listener to the connected callback above and not need to reference the dialog at all

Reply:
It does. I'm sure i added it here for a reason but it seems to work fine in the connected call back now so have moved it 👍